src/EnergyPlus/SolarShading.cc

```diff
@@ -10980,12 +10968,9 @@ void CalcBeamSolarOnWinRevealSurface(EnergyPlusData &state)
             H = state.dataSurface->Surface(SurfNum).Height;
             W = state.dataSurface->Surface(SurfNum).Width;
             d1 = OutsReveal + 0.5 * GlazingThickness;
-            ConstrNum = state.dataSurface->Surface(SurfNum).Construction;
-            ConstrNumSh = state.dataSurface->Surface(SurfNum).activeShadedConstruction;
-            if (state.dataSurface->SurfWinStormWinFlag(SurfNum) == 1) {
-                ConstrNum = state.dataSurface->Surface(SurfNum).StormWinConstruction;
-                ConstrNumSh = state.dataSurface->Surface(SurfNum).activeStormWinShadedConstruction;
-            }
+            ConstrNum = state.dataSurface->SurfWinActiveConstruction(SurfNum);
+            ConstrNumSh = state.dataSurface->SurfWinActiveShadedConstruction(SurfNum);
```

Comment:
Ideally, it should be possible to localize the scope of every instance of `ConstrNum=` and `ConstrNumSh =` (or similar variables). And they should all be `int const ConstrNum = ` Yes?

Reply:
Same here. The logic is to assign `ConstrNum` to the original construction number at the beginning of the loop, but whether to overwrite depends on if the surface is a `Window5DetailedModel` window. Though `ConstrNumSh` is, in general, a const.
Maybe a better way of doing so is to check `if 5DETAILSMODEL && EXTERIOR WINDOW` when assigning. 
```
if (state.dataSurface->SurfWinStormWinFlag(SurfNum) == 1 && 5DETAILSMODEL && ???) {
    state.dataSurface->SurfWinActiveConstruction(SurfNum) = state.dataSurface->SurfWinStormWinConstr(SurfNum);
} else {
    state.dataSurface->SurfWinActiveConstruction(SurfNum) = state.dataSurface->Surface(SurfNum).Construction;
}
```

Reply:
I added the checking of `5DETAILSMODEL` and make most of the assignment local and static.